public/app/features/explore/ExploreToolbar.tsx

```diff
@@ -163,7 +176,7 @@ export function ExploreToolbar({ exploreId, topOfViewRef, onChangeTime }: Props)
             hideTextValue={showSmallDataSourcePicker}
             width={showSmallDataSourcePicker ? 8 : undefined}
           />,
-        ]}
+        ].filter(Boolean)}
```

Comment:
What is this for?

Reply:
since `leftItems` prop is an array, and we are introducing a content outline feature toggle here, which if it's disabled or undefined will still be added as an element to the array and push the sibling button in slightly. So this is just a way to keep only truthy items in the array. here's the example:

Without filter
![Screenshot 2023-09-28 at 1 30 39 PM](https://github.com/grafana/grafana/assets/58232930/0d7f9685-23d9-45a2-989e-dde838721fd2)

With filter
![Screenshot 2023-09-28 at 1 31 30 PM](https://github.com/grafana/grafana/assets/58232930/30e79623-23cb-44a5-8fd2-54f0f1a81e3b)

Reply:
Got it. I know I prefer verbose but clear code more than others so feel free to ignore this, but I'm wondering if it makes more sense to have this be an array we `unshift` to add the outline, and then just pass the array here rather than have this filter. Not necessary but a bit more obvious to me what's going on

Reply:
I've followed the filter(Boolean) pattern because we are doing the same on line 285 with the PageToolbar, although it doesn't seem necessary because it looks like PageToolbar itself takes care of that [here] (https://github.com/grafana/grafana/blob/0af724761227347b2112abd8a177fa1d6e854484/packages/grafana-ui/src/components/PageLayout/PageToolbar.tsx#L148) 🤔. I'll leave it as it is for now because we are using this pattern in other places in the repo, but I agree with you that it's clearer to first create an array, do some checks, and then pass it as a prop.